database/repository/withdraw/withdraw.go

```diff
@@ -0,0 +1,332 @@
+			_ = tx.Rollback()
+			return
+		}
+	}
+	if res.RequestDetails.Type == withdraw.Crypto {
+		cryptoEvent := &modelSQLite.WithdrawalCrypto{
+			Address: res.RequestDetails.Crypto.Address,
+			Fee:     res.RequestDetails.Crypto.FeeAmount,
+		}
+		if res.RequestDetails.Crypto.AddressTag != "" {
+			cryptoEvent.AddressTag.SetValid(res.RequestDetails.Crypto.AddressTag)
+		}
+		err = tempEvent.AddWithdrawalCryptos(ctx, tx, true, cryptoEvent)
+		if err != nil {
+			log.Errorf(log.DatabaseMgr, "Event Insert failed: %v", err)
+			_ = tx.Rollback()
+			return
+		}
+	}
+	res.ID = newUUID
+	return nil
+}
+
+// TODO: optimize these queries
+
+// EventByUUID return requested withdraw information by ID
+func EventByUUID(id string) (*withdraw.Response, error) {
+	if database.DB.SQL == nil {
+		return nil, errors.New("database is nil")
+	}
+
+	var resp = &withdraw.Response{}
+	var ctx = context.Background()
+
+	if repository.GetSQLDialect() == database.DBSQLite3 {
+		v, err := modelSQLite.FindWithdrawalHistory(ctx, database.DB.SQL, id, "*")
+		if err != nil {
+			return nil, err
+		}
+		newUUID, _ := uuid.FromString(v.ID)
+		resp.ID = newUUID
+		resp.Exchange = new(withdraw.ExchangeResponse)
+		resp.Exchange.ID = v.ExchangeID
+		resp.Exchange.Name = v.Exchange
+		resp.Exchange.Status = v.Status
+		resp.RequestDetails = new(withdraw.Request)
+		resp.RequestDetails = &withdraw.Request{
+			Currency:    currency.Code{},
+			Description: v.Description.String,
+			Amount:      v.Amount,
+			Type:        withdraw.RequestType(v.WithdrawType),
+		}
+
+		createdAtTime, err := time.Parse("2006-01-02T15:04:05Z", v.CreatedAt)
+		if err != nil {
+			log.Errorf(log.DatabaseMgr, "time conversion error defaulting to empty time: %v", err)
+			resp.CreatedAt = time.Time{}
+		} else {
+			resp.CreatedAt = createdAtTime.UTC()
+		}
+		updatedAtTime, err := time.Parse("2006-01-02T15:04:05Z", v.UpdatedAt)
+		if err != nil {
+			log.Errorf(log.DatabaseMgr, "time conversion error defaulting to empty time: %v", err)
+			resp.UpdatedAt = time.Time{}
+		} else {
+			resp.UpdatedAt = updatedAtTime.UTC()
+		}
+	} else {
+		v, err := modelPSQL.FindWithdrawalHistory(ctx, database.DB.SQL, id, "*")
+		if err != nil {
+			return nil, err
+		}
+
+		newUUID, _ := uuid.FromString(v.ID)
+		resp.ID = newUUID
+		resp.Exchange = new(withdraw.ExchangeResponse)
+		resp.Exchange.ID = v.ExchangeID
+		resp.Exchange.Name = v.Exchange
+		resp.Exchange.Status = v.Status
+		resp.RequestDetails = new(withdraw.Request)
+		resp.RequestDetails = &withdraw.Request{
+			Currency:    currency.Code{},
+			Description: v.Description.String,
+			Amount:      v.Amount,
+			Type:        withdraw.RequestType(v.WithdrawType),
+		}
+		resp.CreatedAt = v.CreatedAt
+		resp.UpdatedAt = v.UpdatedAt
+
+		if withdraw.RequestType(v.WithdrawType) == withdraw.Crypto {
+			resp.RequestDetails.Crypto = new(withdraw.CryptoRequest)
+			x, err := v.WithdrawalCryptoWithdrawalCryptos().One(ctx, database.DB.SQL)
```

Comment:
[One()](https://github.com/xtda/gocryptotrader/blob/8d19edb0e498651f6607b5f743ea11220ebf9919/database/models/postgres/withdrawal_crypto.go#L313) initialises an instance of WithdrawalCrypto and returns a pointer to it (or error if anything inside returns an error)

The error check and return on the next line is enough to catch this
There is about as much chance as it also being nil after a 
```go
if x == nil {}
``` 
check 

And if this happens its a sign of even bigger problems

Reply:
I like it as a defensive guarantee, but I'm going to resolve this because think I'm being too defensive of the panic. Figure I'd explain my reasoning though. 
`One()` could change at any time and this function would have no oversight. It puts a burden on future reviewers to ensure that this still operates properly even if this specific code doesn't get modified. The only way to guarantee that its not nil, is by checking it where you're going to use it